### ISSUE TITLE
[Feature] Support for multi-line yaml anchors in config

### ISSUE THREAD

**Issue by aleksandarmomic** [joshuaavalon/docker-cloudflare#59]
Because domain entries are mostly repetitive, it would be nice to support multi-line anchors to avoid copy pasting.
When using multi-line yaml anchors, the schema validator throws errors in logs and keeps trying to validate the yaml.
I created a `default` anchor to be used everywhere and in the second domain I just override the name which is valid yaml, but does not get parsed by the app.

This is the example config I made:
```yaml
auth:
  scopedToken: <scoped-token>
domains:
  - &default
    name: <domain>
    type: A
    proxied: false
    create: true
    zoneId: <zone-id>

  - <<: *default
    name: <domain2>
```

The error that gets thrown:
```yaml
{
  data: {
    auth: { scopedToken: '<scoped-token>' },
    domains: [ [Object], [Object] ]
  },
  schema: {
    '$id': 'https://joshuaavalon.github.io/docker-cloudflare/config.schema.json',
    '$schema': 'http://json-schema.org/draft-07/schema#',
    description: 'Configuration for docker-cloudflare',
    type: 'object',
    properties: {
      api: [Object],
      logLevel: [Object],
      auth: [Object],
      domains: [Object],
      ipv4: [Object],
      ipv6: [Object],
      echoParsers: [Object]
    },
    required: [ 'auth', 'domains' ],
    additionalProperties: false,
    definitions: {
      globalAuth: [Object],
      scopedAuth: [Object],
      domain: [Object],
      ipEcho: [Object],
      ipEchoParser: [Object],
      webhook: [Object]
    }
  }
}

/app/packages/schema/lib/index.js:37
    throw new SchemaViolationError(validate.errors || []);
          ^

SchemaViolationError: Failed to validate schema Errors:
must NOT have additional properties
    at verifySchema (/app/packages/schema/lib/index.js:37:11)
    at readFileConfig (/app/packages/config/lib/file-config.js:30:47)
    at async readConfig (/app/packages/config/lib/index.js:21:24)
    at async main (/app/packages/app/lib/index.js:80:20) 
    {
  errors: [
    {
      instancePath: '/domains/1',
      schemaPath: '#/additionalProperties',
      keyword: 'additionalProperties',
      params: { additionalProperty: '<<' },
      message: 'must NOT have additional properties'
    }
  ]
}
```

I think the solution is obvious, schema needs to be updated.

As a side note:
The schema id in the logs is not a valid url, maybe it just needs to be updated to the new location? 
- [https://github.com/joshuaavalon/docker-cloudflare/blob/master/packages/config/src/config.schema.json](https://github.com/joshuaavalon/docker-cloudflare/blob/master/packages/config/src/config.schema.json)

When I throw in to vscode to use the valid schema url, it doesn't complain.

**Comment by joshuaavalon** [comment]
For the syntax `- <<: *default` , merge only exists in YAML 1.1 which is obsoleted and it is not included in YAML 1.2. While I use `cosmiconfig` to parse config, `cosmiconfig` use `yaml` to parse YAML config.

`yaml` supports merge behind a [feature flag](https://eemeli.org/yaml/#schema-options) by setting `merge` to `true`. However, `cosmiconfig` does not provide this option. You should open a issue at [`cosmiconfig`](https://github.com/davidtheclark/cosmiconfig) for enabling feature flags for `yaml`. Currently, there is no way for me to enable this feature without forking `cosmiconfig`.

For the JSON schema, it is intended for internal use, so the `$id` does not matter. But I will publish it via GitHub page to make it valid.

**Comment by joshuaavalon** [comment]
https://joshuaavalon.github.io/docker-cloudflare/config.schema.json now works.

**Comment by joshuaavalon** [comment]
Sorry if you already know it, but you can use **CNAME** instead of **A** record. For example, you want to point 

* `example.com`
* `foo.example.com`
* `bar.example.com`

You can create **CNAME** record for `foo.example.com` and `bar.example.com` and point to `example.com`. Now you only need to update `example.com`.

**Comment by aleksandarmomic** [comment]
> For the syntax `- <<: *default` , merge only exists in YAML 1.1 which is obsoleted and it is not included in YAML 1.2. While I use `cosmiconfig` to parse config, `cosmiconfig` use `yaml` to parse YAML config.
> 
> `yaml` supports merge behind a [feature flag](https://eemeli.org/yaml/#schema-options) by setting `merge` to `true`. However, `cosmiconfig` does not provide this option. You should open a issue at [`cosmiconfig`](https://github.com/davidtheclark/cosmiconfig) for enabling feature flags for `yaml`. Currently, there is no way for me to enable this feature without forking `cosmiconfig`.
> 
> For the JSON schema, it is intended for internal use, so the `$id` does not matter. But I will publish it via GitHub page to make it valid.

Understood, thank you for such detailed answer!



> Sorry if you already know it, but you can use **CNAME** instead of **A** record. For example, you want to update
> 
>     * `example.com`
> 
>     * `foo.example.com`
> 
>     * `bar.example.com`
> 
> 
> You can create **CNAME** records for `foo.example.com` and `bar.example.com` and point them to `example.com`. Now you only need to update `example.com`.

Yeah, I've set up my root domain to use argo tunnel and all things go through it with CNAMEs pointing to it. I have a special case here for your container because I have two subdomains that HAVE TO bypass the argo tunnel because they are udp, which is not supported by the argo tunnel currently. These two are getting updated via A records. I guess I could create one record that will have the ip, and just update that with other two pointing to it, but opted for current setup.
